pilot/pkg/model/validation.go

```diff
@@ -1540,9 +1540,10 @@ func checkServiceRoleBinding(in *rbac.ServiceRoleBinding) error {
 			errs = appendErrors(errs, fmt.Errorf("do not use * for names or not_names (in rule %d)", i))
 		}
 	}
-	if in.RoleRef == nil {
-		errs = appendErrors(errs, fmt.Errorf("roleRef must be specified"))
-	} else {
+	if in.RoleRef == nil && len(in.Actions) == 0 {
+		errs = appendErrors(errs, fmt.Errorf("`roleRef` or `actions` must be specified"))
+	}
+	if in.RoleRef != nil {
```

Comment:
We should validate and return error if both roleRef and action are used.

Reply:
Good catch. Thanks!